packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -380,7 +419,17 @@ class UserSessionManager implements IUserSessionManager {
         storageType === COOKIE_STORAGE
       ) {
         // remove cookie that is set from server side
-        this.setServerSideCookie([{ name: key, value: '' }], curStore);
+        this.setServerSideCookie(
+          [{ name: key, value: '' }],
+          (cookieName, cookieValue) => {
```

Comment:
Here, we know the context that this is a request to delete the cookie. Then why does the callback also handle the `set` part?

Reply:
Similar to the above callback function.

Reply:
Addressed